--- a/toolchain/internal/configure.bzl
+++ b/toolchain/internal/configure.bzl
@@ -347,6 +347,7 @@ def _cc_toolchain_str(
         "darwin-aarch64": "aarch64-apple-macosx",
         "linux-aarch64": "aarch64-unknown-linux-gnu",
         "linux-armv7": "armv7-unknown-linux-gnueabihf",
+        "linux-riscv64": "riscv64-unknown-linux-gnu",
         "linux-x86_64": "x86_64-unknown-linux-gnu",
         "none-riscv32": "riscv32-unknown-none-elf",
         "none-x86_64": "x86_64-unknown-none",
