--- a/base/cvd/cuttlefish/bazel/rules.bzl	2026-09-10 21:46:27.737014550 +0000
+++ b/base/cvd/cuttlefish/bazel/rules.bzl	2026-09-10 21:47:01.305890411 +0000
@@ -17,13 +17,11 @@
 invocation targets.
 """
 
-load("@aspect_rules_lint//format:defs.bzl", "format_test")
-load("@cc_compatibility_proxy//:proxy.bzl", "cc_binary", "cc_library", "cc_test")
+load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library", "cc_test")
 load("@rules_rust//rust:defs.bzl", "rust_binary")
 load("@rules_shell//shell:sh_binary.bzl", "sh_binary")
 load("@rules_shell//shell:sh_library.bzl", "sh_library")
 load("//:build_variables.bzl", BUILD_VAR_COPTS = "COPTS", BUILD_VAR_LINKOPTS = "LINKOPTS")
-load("//tools/lint:linters.bzl", "buildifier_test", "clang_tidy_include_cleaner_test", "clang_tidy_test", "dwyu_rule", "shellcheck_test")
 
 visibility(["//..."])
 
@@ -49,22 +47,12 @@
 macro = _fallback_macro
 
 def _cf_build_test_implementation(name, srcs, **kwargs):
-    native.filegroup(
-        name = name + "_LINT_TEST_starlark_files",
-        srcs = (srcs or []) + ["BUILD.bazel"],
-        tags = ["lint-with-buildifier"],
-    )
-    buildifier_test(
-        name = name + "_LINT_TEST",
-        srcs = [":" + name + "_LINT_TEST_starlark_files"],
-        **kwargs
-    )
+    pass
 
 cf_build_test = macro(
     inherit_attrs = "common",
     attrs = {
         "srcs": attr.label_list(
-            configurable = False,
             default = [],
         ),
     },
@@ -83,7 +71,7 @@
         linkopts,
         _target_type,
         **kwargs):
-    if not clang_tidy_enabled and not kwargs["deprecation"]:
+    if not clang_tidy_enabled and not kwargs.get("deprecation"):
         kwargs["deprecation"] = "Not covered by clang-tidy"
 
     if _target_type == "cc_binary":
@@ -119,68 +107,32 @@
     else:
         fail("Unknown target type " + _target_type)
 
-    if clang_format_enabled and (kwargs.get("srcs") or kwargs.get("hdrs")):
-        format_test(
-            name = name + "_format_test",
-            cc = "//tools/format:clang_format",
-            disable_git_attribute_checks = True,
-            srcs = (kwargs.get("srcs") or []) + (kwargs.get("hdrs") or []),
-            visibility = ["//visibility:private"],
-        )
-    if clang_tidy_enabled:
-        if include_cleaner_enabled:
-            clang_tidy_include_cleaner_test(
-                name = name + "_clang_tidy",
-                srcs = [":" + name],
-                tags = ["clang_tidy", "clang-tidy"],
-                visibility = ["//visibility:private"],
-            )
-        else:
-            clang_tidy_test(
-                name = name + "_clang_tidy",
-                srcs = [":" + name],
-                tags = ["clang_tidy", "clang-tidy"],
-                visibility = ["//visibility:private"],
-            )
-    if depend_on_what_you_use_enabled and not "-layering_check" in features:
-        dwyu_rule(
-            name = name + "_depend_on_what_you_use",
-            deps = [":" + name],
-            testonly = True,
-            visibility = ["//visibility:private"],
-        )
+
 
 _CC_COMMON_ATTRS = {
     "clang_format_enabled": attr.bool(
-        configurable = False,
-        default = True,
+        default = False,
         doc = "Decide if a corresponding format_test target is generated",
     ),
     "clang_tidy_enabled": attr.bool(
-        configurable = False,
-        default = True,
+        default = False,
         doc = "Decide if a corresponding clang_tidy_test target is generated",
     ),
     "copts": attr.string_list(
-        configurable = False,
         default = [],
     ),
     "depend_on_what_you_use_enabled": attr.bool(
-        configurable = False,
-        default = True,
+        default = False,
         doc = "Decide if a corresponding depend-on-what-you-use target is generated",
     ),
     "features": attr.string_list(
-        configurable = False,
         default = [],
     ),
     "include_cleaner_enabled": attr.bool(
-        configurable = False,
-        default = True,
+        default = False,
         doc = "Run clang-tidy with misc-include-cleaner",
     ),
     "linkopts": attr.string_list(
-        configurable = True,
         default = [],
     ),
 }
@@ -188,8 +140,8 @@
 cf_cc_binary = macro(
     inherit_attrs = cc_binary,
     attrs = _CC_COMMON_ATTRS | {
-        "deps": attr.label_list(configurable = True),
-        "_target_type": attr.string(configurable = False, default = "cc_binary"),
+        "deps": attr.label_list(),
+        "_target_type": "cc_binary",
     },
     implementation = _cf_cc_target_implementation,
 )
@@ -197,8 +149,8 @@
 cf_cc_library = macro(
     inherit_attrs = cc_library,
     attrs = _CC_COMMON_ATTRS | {
-        "deps": attr.label_list(configurable = True),
-        "_target_type": attr.string(configurable = False, default = "cc_library"),
+        "deps": attr.label_list(),
+        "_target_type": "cc_library",
     },
     implementation = _cf_cc_target_implementation,
 )
@@ -206,8 +158,8 @@
 cf_cc_test = macro(
     inherit_attrs = cc_test,
     attrs = _CC_COMMON_ATTRS | {
-        "deps": attr.label_list(configurable = False),
-        "_target_type": attr.string(configurable = False, default = "cc_test"),
+        "deps": attr.label_list(),
+        "_target_type": "cc_test",
     },
     implementation = _cf_cc_target_implementation,
 )
@@ -217,13 +169,6 @@
         name = name,
         **kwargs
     )
-    format_test(
-        name = name + "_format_test",
-        rust = "@rules_rust//tools/upstream_wrapper:rustfmt",
-        disable_git_attribute_checks = True,
-        srcs = (kwargs.get("srcs") or []),
-        visibility = ["//visibility:private"],
-    )
 
 cf_rust_binary = macro(
     inherit_attrs = rust_binary,
@@ -235,18 +180,11 @@
         name = name,
         **kwargs
     )
-    if shellcheck_enabled:
-        shellcheck_test(
-            name = name + "_shellcheck",
-            srcs = [":" + name],
-            tags = ["shellcheck"],
-            visibility = ["//visibility:private"],
-        )
 
 cf_sh_binary = macro(
     inherit_attrs = sh_binary,
     attrs = {
-        "shellcheck_enabled": attr.bool(configurable = False, default = True, doc = "Decide if a corresponding shellcheck_test target is generated"),
+        "shellcheck_enabled": attr.bool(default = False, doc = "Decide if a corresponding shellcheck_test target is generated"),
     },
     implementation = _cf_sh_binary_implementation,
 )
@@ -256,18 +194,11 @@
         name = name,
         **kwargs
     )
-    if shellcheck_enabled:
-        shellcheck_test(
-            name = name + "_shellcheck",
-            srcs = [":" + name],
-            tags = ["shellcheck"],
-            visibility = ["//visibility:private"],
-        )
 
 cf_sh_library = macro(
     inherit_attrs = sh_library,
     attrs = {
-        "shellcheck_enabled": attr.bool(configurable = False, default = True, doc = "Decide if a corresponding shellcheck_test target is generated"),
+        "shellcheck_enabled": attr.bool(default = False, doc = "Decide if a corresponding shellcheck_test target is generated"),
     },
     implementation = _cf_sh_library_implementation,
 )
