diff --git a/BUILD.bazel b/BUILD.bazel
index d920390..44c9b13 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -1,3 +1,4 @@
+load("@rules_cc//cc:defs.bzl", "cc_library")
 load("@bazel_skylib//rules:expand_template.bzl", "expand_template")
 
 SUBSTITUTIONS = {
diff --git a/MODULE.bazel b/MODULE.bazel
index 45f448e..8f14f11 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,8 +1,9 @@
 module(
     name = "libjpeg_turbo",
     version = "2.1.91",
     compatibility_level = 1,
 )
 
 bazel_dep(name = "bazel_skylib", version = "1.5.0")
 bazel_dep(name = "platforms", version = "0.0.8")
+bazel_dep(name = "rules_cc", version = "0.0.9")
