diff --git a/third_party/marl/BUILD.bazel b/third_party/marl/BUILD.bazel
index 297fd4464..221985901 100644
--- a/third_party/marl/BUILD.bazel
+++ b/third_party/marl/BUILD.bazel
@@ -12,6 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@rules_cc//cc:cc_library.bzl", "cc_library")
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+
 config_setting(
     name = "linux_x86_64",
     constraint_values = [
