diff --git a/sandboxed_api/bazel/proto.bzl b/sandboxed_api/bazel/proto.bzl
index bd36c886..77d0a5ee 100644
--- a/sandboxed_api/bazel/proto.bzl
+++ b/sandboxed_api/bazel/proto.bzl
@@ -15,6 +15,7 @@
 """Generates proto targets in various languages."""
 
 load("@com_google_protobuf//bazel:proto_library.bzl", "proto_library")
+load("@com_google_protobuf//bazel:cc_proto_library.bzl", "cc_proto_library")
 load("@rules_cc//cc:cc_library.bzl", "cc_library")
 
 def _cc_proto_library_name_from_proto_name(name):
@@ -75,7 +76,7 @@ def sapi_proto_library(
         srcs = srcs,
         deps = deps,
     )
-    native.cc_proto_library(
+    cc_proto_library(
         name = name + "_sapi_cc_proto",
         deps = [name],
     )
